### ISSUE TITLE
Encrypted rooms often show red shield despite all sessions trusted & users verified

### ISSUE THREAD

**Issue by krassle** [vector-im/element-android#8046]
### Steps to reproduce

1. Be in an e2ee room with 3+ people, where all participants are verified and their sessions trusted
2. Observe small red shield icon next to room avatar
3. Verify all participants have no untrusted sessions
4. Navigate to Settings -› Security & Privacy -› Active Sessions -› Show All Sessions
5. Then go back to room and observe: *red shield* icon changed to *green* with a checkmark!

[1/6]
________________________________________________
<img src="https://user-images.githubusercontent.com/6473406/215793151-e12fe47d-2d40-45f8-bec6-dd55ea2590cf.jpg" width="300" />

________________________________________________
[2/6]
<img src="https://user-images.githubusercontent.com/6473406/215793225-ce1a8036-4e9d-424e-9f6f-e49fc06d37a3.jpg" width="300" />

________________________________________________
[3/6]
<img src="https://user-images.githubusercontent.com/6473406/215793337-3705e50f-3519-442b-acbe-3e9a3926550b.jpg" width="300" />

________________________________________________
[4/6]
<img src="https://user-images.githubusercontent.com/6473406/215793559-f6e8e28f-a189-4cef-9b75-1a0c3e26cb3f.jpg" width="300" />

________________________________________________
[5/6]
<img src="https://user-images.githubusercontent.com/6473406/215793608-222aa56a-76b7-4a6f-8bfb-2e3a2ed47707.jpg" width="300" />

________________________________________________
[6/6]
<img src="https://user-images.githubusercontent.com/6473406/215793828-7a2c14ce-e2b5-4864-89ce-2b11f0094cf4.jpg" width="300" />

________________________________________________
This bug exists since at least  v.1.5.18, but does not affect Element-Web or Desktop. 

#### What did you expect?
No *random* status change of e2e encrypted rooms  (verified/trusted->unverified/untrusted)
#### What happened instead?
Red shield icon despite all users verified and their sessions trusted and trusted/verified Green shield icon shown only after manual intervention 

### Your phone model

HUAWEI LLD-L31

### Operating system version

Android 9

### Application version and app store

Element version 1.5.22 [40105222] (G-bdd431d2), olm version 3.2.12 from Google Play Store

### Homeserver

Synapse 1.73

### Will you send logs?

No

### Are you willing to provide a PR?

No

**Comment by TheCycoONE** [comment]
I just hit something like this. The room had a red shield and the people view had a red shield on me but all my sessions were trusted. Went away on restarting the app.
![Screenshot_20230216-232349_Element](https://user-images.githubusercontent.com/823433/219549933-e98b2bc1-5e92-4c6e-bd43-528a0dbd28c9.jpg)

**Comment by yostyle** [comment]
![image](https://user-images.githubusercontent.com/1191674/220279773-802729fd-512a-4a53-821a-ae0d1bbd1b03.png)

In this screen the shield at the left of the username is green if all devices are verified by **cross signing only** otherwise the shield is red (or black if the current session is not verified).

The shields in the security section are green if all devices are verified **locally** or by **cross signing**.

So the state of your screenshot is possible if you verify a session without cross signing or if the current session state is not refresh properly after a verification by cross signing

**Comment by krassle** [comment]
@yostyle 
>So the state of your screenshot is possible if you verify a session without cross signing or if the current session state is not refreshed properly after a verification by cross signing.

My observation(as a hint):
The *state* changes from red to green (screen **[2/6] --›› [6/6]**) immediately after viewing **Show All Sessions** (Settings -› Security & Privacy -› Active Sessions -› Show All Sessions). So probably, as you suggested, it is not properly refreshed (or read from cache or something) on every app startup, therefore appears randomly and is also mitigated by an app restart as mentioned by @TheCycoONE.

But I don't think this *state* should be possible(intentional or valid) at all, as it'd be extremely confusing showing a *verified* user with all *trusted* sessions, but with an alarming red shield next to the username. What is the issue then and where is the distinction? :confused: 

>I can't reproduce this issue and I need to investigate more time to understand what's happen.

Thank you for taking the time to look into this, it is very much appreciated. :+1: :wink: 

Please let me also know if I can help in any other way, as on my side this issue is almost always reproducible (on every second or third run of the app). So far I've tried clearing the cache (Settings -› General -› Advanced -› Clear cache) but without success...

**Comment by TheCycoONE** [comment]
I saw the red shield on the room again today, and again looking at the list of users it was on me. I had not added any new sessions since last time and again it went green on restart.

**Comment by Gitogitos** [comment]
Have exactly the same, all my sessions are verified, trusted  but when click on my user I got red shield. Cross signing is activated the same. My version of Element is 1.5.22 [4010522] (G-bdd431d2), This could be related to this but it’s strange cause she is verified as well on every session.
![Screenshot_20230301-140556](https://user-images.githubusercontent.com/101456208/222165468-f8e25c6b-baf8-428f-aeeb-258d79a02e52.png)
![Screenshot_20230301-140614](https://user-images.githubusercontent.com/101456208/222165508-d2ee6c67-f7b9-4e7c-aa3d-a3281ca624ee.png)

**Comment by giomfo** [comment]
I will close this issue because the context has changed since PR #8195. Indeed the room member shields should be coherent in the different screen.
This change has been released in [v1.5.28](https://github.com/vector-im/element-android/releases/tag/v1.5.28)

Please create a new issue if you still observe a **persistent** unexpected red shields. Logs (Rageshakes) would be then very useful.

**Comment by krassle** [comment]
Still not fixed in v1.5.28. Rageshake sent.

**Comment by yostyle** [comment]
Do you have the same result in room member screen ? it shouldn't.

<img src="https://user-images.githubusercontent.com/6473406/215793337-3705e50f-3519-442b-acbe-3e9a3926550b.jpg" width="300" />

**Comment by giomfo** [comment]
@krassle thanks for your new screenshots
FYI I created a [new issue](https://github.com/vector-im/element-android/issues/8308), I will ask to the crypto team to work on it

**Comment by BillCarsonFr** [comment]
@krassle if you can reproduce can you go the Settings > Advanced Settings > Enable verbose logs and send a new RS
